tensorbay/utility/common.py

```diff
@@ -122,3 +123,66 @@ def _update_docstring(self, docstring: Optional[str]) -> str:
         lines.insert(1, "")
 
         return "\n".join(lines)
+
+
+class KwargsDeprecated:  # pylint: disable=too-few-public-methods
+    """A decorator for the function which has deprecated keyword argument.
+
+    Arguments:
+        since: The version the keyword arguments are deprecated.
+        remove_in: The version the keyword arguments will be removed in.
+        substitute: The substitute usage.
+
+    """
+
+    def __init__(
+        self,
+        keywords: Tuple[str, ...],
+        *,
+        since: str,
+        removed_in: Optional[str] = None,
+        substitute: Optional[str] = None,
+    ) -> None:
+        self._keywords = keywords
+        self._since = since
+        self._removed_in = removed_in
+        self._substitute = substitute
+
+    def __call__(self, func: _Callable) -> _Callable:
+        """Wrap the decorated function by adding the deprecated message.
+
+        Arguments:
+            func: The deprecated function.
+
+        Returns:
+            The wrapped function which shows the deprecated message when calling.
+
+        """
+        if len(self._keywords) == 1:
+            keyword_message = self._keywords[0]
+        else:
+            keyword_message = " ".join(f'"{keyword}"' for keyword in self._keywords[:-1])
+            keyword_message = f'{keyword_message} and "{self._keywords[-1]}"'
+
+        messages = [
+            (
+                f'The keyword arguments: {keyword_message} in "{func.__name__}" '
```

Comment:
Is it necessary that the message change based on different "keyword_message"?
"The keyword argument ... is ..." and "The keyword arguments ... are ..."?